File: leetcode/problems/go/0045. Jump Game II/solve.go
package main

func jump(nums []int) int {
	L := len(nums)
	if L == 1 {
		return 0
	}
	steps := 1
	farthest := 0
	checkEnd := 0
	for i, n := range nums {
		farthest = max(i+n, farthest)
		if farthest >= L-1 {
			break
		}
		if i == checkEnd {
			steps += 1
			checkEnd = farthest
		}
	}
	return steps
}

File: leetcode/problems/typescript/0045. Jump Game II/solve.ts
function jump(nums: number[]): number {
  const N = nums.length;
  if (N <= 1) {
    return 0;
  }

  let steps = 1;
  let farthest = 0;
  let checkEnd = 0;

  for (let i = 0; i < N - 1; i++) {
    farthest = Math.max(i + nums[i], farthest);
    if (farthest >= N - 1) {
      break;
    }
    if (i == checkEnd) {
      steps += 1;
      checkEnd = farthest;
    }
  }
  return steps;
}
